src/arc.rs

```diff
@@ -149,6 +149,40 @@ impl<T> Arc<T> {
         mem::forget(a);
         unsafe { Arc::from_raw(ptr) }
     }
+
+    /// Returns the inner value, if the [`Arc`] has exactly one strong reference.
+    ///
+    /// Otherwise, an [`Err`] is returned with the same [`Arc`] that was
+    /// passed in.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use triomphe::Arc;
+    ///
+    /// let x = Arc::new(3);
+    /// assert_eq!(Arc::try_unwrap(x), Ok(3));
+    ///
+    /// let x = Arc::new(4);
+    /// let _y = Arc::clone(&x);
+    /// assert_eq!(*Arc::try_unwrap(x).unwrap_err(), 4);
+    /// ```
+    pub fn try_unwrap(this: Self) -> Result<T, Self> {
```

Comment:
I wonder if this should be returning a UniqueArc, and then we have an unwrap on that type?

Or at least have a `try_unique()` that is used by `try_unwrap()`

Reply:
I added `try_unique()`, good idea!